### PR TITLE
Added JsonByteWriter

### LINKED ISSUE #9: Write JSON directly to utf8 encoded bytes?
Something like: `jsonStringWriter`, but maybe called `jsonUtf8Writer(StreamSink<List<int>> byteSink)` would be really nice.

I'm currently doing:
```dart
final out = StringBuffer();
final w = jsonStringWriter(out);

w.startObject();
// ...

w.endObject();

final bytes = utf8.encoder.convert(out.toString());
```
Maybe, there is a better way that I'm just not seeing? or a feature for writing directly to utf8 would be nice :D

---------

For discoverability it might also be nice to put these things as static functions on `JsonWriter`, like:
 * `JsonWriter.utf8Writer(StreamSink<List<int>> byteSink, {String? indent, bool asciiOnly = false})`
 * `JsonWriter.stringWriter(StringSink sink, {String? indent, bool asciiOnly = false})`

Instead of having them float like top-level functions.

### PR DESCRIPTION
**Pull request by jonasfj**
Added `jsonByteWriter` for writing to a `Sink<List<int>>` directly.

Fixes https://github.com/lrhn/json-tool/issues/9

### CODE REVIEW

**Comment by jonasfj** [comment]
Also included a fix in `JsonObjectWriter`.

File: lib/src/json/sink/byte_writer.dart [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+  /// whitespace between tokens.
+  JsonByteWriter(
+    Sink<List<int>> target, {
+    Encoding encoding = utf8,
+    bool? asciiOnly,
+  })  : _encoding = encoding,
+        _target = target,
+        _sink = encoding.encoder
+            .startChunkedStringConversion(_NonClosingSink(target)),
+        _asciiOnly = asciiOnly ?? encoding != utf8;
```
Comment (lrhn) [inline lib/src/json/sink/byte_writer.dart:65]: Not optimal for `latin1`.

File: lib/src/json/sink/object_writer.dart [hunk 2]
```diff
@@ -96,7 +96,7 @@ class JsonObjectWriter implements JsonWriter<Object?> {
   void startArray() {
     var array = <dynamic>[];
     _stack.add(_key);
-    _stack.add(<dynamic>[]);
+    _stack.add(array);
```
Comment (lrhn) [inline lib/src/json/sink/object_writer.dart:99]: Well, doh!

**Review by lrhn — COMMENTED** [review]
I'll give it a once-over after landing.

**Pull request merged**